src/Shape.js

```diff
@@ -386,48 +386,33 @@
             var threshold = alphaThreshold || 0,
                 cachedCanvas = this._cache.canvas,
                 sceneCanvas = this._getCachedSceneCanvas(),
-                sceneContext = sceneCanvas.getContext(),
                 hitCanvas = cachedCanvas.hit,
                 hitContext = hitCanvas.getContext(),
-                width = sceneCanvas.getWidth(),
-                height = sceneCanvas.getHeight(),
-                sceneImageData, sceneData, hitImageData, hitData, len, rgbColorKey, i, alpha;
+                hitWidth = hitCanvas.getWidth(),
+                hitHeight = hitCanvas.getHeight(),
+                hitImageData, hitData, len, rgbColorKey, i, alpha;
 
             hitContext.clear();
-            var ratio = sceneCanvas.pixelRatio;
-            var hitWidth = Math.floor(width / ratio);
-            var hitHeight = Math.floor(height / ratio);
+            hitContext.drawImage(sceneCanvas._canvas, 0, 0, hitWidth, hitHeight);
 
             try {
-                // if pixelRatio > 1 we need to redraw scene canvas into temp canvas
-                // with pixelRatio = 1 to have the same dimension as hitCanvas
-                if (ratio !== 1) {
-                    var tempCanvas = new Konva.SceneCanvas({
-                        width : hitWidth,
-                        height : hitHeight,
-                        pixelRatio : 1
-                    });
-                    tempCanvas.getContext().drawImage(sceneCanvas._canvas, 0, 0, hitWidth, hitHeight);
-                    sceneImageData = tempCanvas.getContext().getImageData(0, 0, hitWidth, hitHeight);
-                    sceneData = sceneImageData.data;
-                } else {
-                    sceneImageData = sceneContext.getImageData(0, 0, width, height);
-                    sceneData = sceneImageData.data;
-                }
                 hitImageData = hitContext.getImageData(0, 0, hitWidth, hitHeight);
```

Comment:
It does not make any difference because code below effectively replaces image data. See "else" branch which lowers alpha to 0.